casepro/msgs/views.py

```diff
@@ -196,10 +213,52 @@ def get_context_data(self, **kwargs):
 
         def render_to_response(self, context, **response_kwargs):
             return JsonResponse({
-                'results': [m.as_json() for m in context['object_list']],
+                'results': [m.as_json(context.get('user_id')) for m in context['object_list']],
                 'has_more': context['has_more']
             }, encoder=JSONEncoder)
 
+    class Touch(OrgPermsMixin, SmartTemplateView):
+        """
+        AJAX endpoint for updating messages with a date and user id.
+        Takes a list of message ids.
+        """
+        @classmethod
+        def derive_url_pattern(cls, path, action):
+            return r'^message/touch/(?P<action>\w+)/$'
+
+        def post(self, request, *args, **kwargs):
+            org = request.org
+            user = request.user
+
+            action = kwargs['action']
+
+            message_ids = request.json['messages']
+            messages = org.incoming_messages.filter(org=org, backend_id__in=message_ids)
+
+            busy_messages = []
+
+            if action == 'busy':
+                for message in messages:
+                    if message.get_busy(request.user.id):
+                        busy_messages.append(message.backend_id)
+
+                if not busy_messages:
+                    for message in messages:
+                        message.last_action = timezone.now()
+                        message.actioned_by = user
+                        message.save()
```

Comment:
Does this mean that we want this api call to be a no-op if there are already busy messages in the set of messages given? I would have expected us to just ignore busy messages but still change messages that are not yet busy.